### PR TITLE
Correct & clarify parts of API spec

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:

GET  /status endpoint: Correct response schema for model_id -  a string, not an object.

GET /conversations/{conversation_id}/tracker : Describe each of the enum options for include_events query parameter

POST & PUT /conversations/{conversation_id}/tracker/events : Events schema added for each event type

GET /conversations/{conversation_id}/story: Clarified  the `all_sessions` query parameter and default behaviour.
Also fixed the story returned when a session is ended by a `restarted` event, so that the story is not blank.

POST /model/test/intents :  Remove JSON option, since the server was interpreting all input as YAML regardless of content-type header.

POST /model/parse: Explain what emulation_mode is and how it affects response results

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/http-api.mdx [hunk 1]
```diff
@@ -50,54 +50,44 @@ The [SocketIO channel](./connectors/your-own-website.mdx#websocket-channel) does
 
 ## Security Considerations
 
-We recommend to not expose the Rasa Server to the outside world, but
-rather connect to it from your backend over a private connection (e.g.
-between docker containers).
+We recommend that you don't expose the Rasa Server to the outside world directly, but
+rather connect to it from via e.g. Nginx as a reverse proxy.
```
Comment (melindaloubser1) [inline docs/docs/http-api.mdx:54]:
```suggestion
rather connect to it via e.g. Nginx.
```

File: rasa/shared/core/trackers.py [hunk 2]
```diff
@@ -940,3 +940,33 @@ def get_trackers_for_conversation_sessions(
...
+
+    Returns:
+        The tracker up until the first restart event.
+    """
+    split_conversations = events.split_events(
+        tracker.events,
+        Restarted,
+        include_splitting_event=False,
+    )
+    if len(split_conversations) == 0:
```
Comment (ancalita) [inline rasa/shared/core/trackers.py:963]: We need some unit test coverage for this new function - I wasn't familiar with this `split_events` function, but it seems that it will always return a `List[List[Event]]` if there are events in the tracker, regardless whether there is a restart or not. The only time it will return an empty list is when the tracker has no events 🤔

File: rasa/server.py [hunk 3]
```diff
@@ -1152,13 +1158,11 @@ async def evaluate_intents(
...
-            if cross_validation_folds:
-                raise ErrorResponse(
-                    HTTPStatus.BAD_REQUEST,
-                    "TestingError",
-                    "Cross-validation is only supported for YAML data.",
-                )
+            raise ErrorResponse(
+                HTTPStatus.BAD_REQUEST,
+                "TestingError",
+                "NLU evaluation is only supported for YAML data.",
```
Comment (ancalita) [inline rasa/server.py:1164]: Why is JSON removed as option?

File: rasa/server.py [hunk 4]
```diff
@@ -317,6 +321,8 @@ async def get_test_stories(
     else:
         trackers = [await processor.get_tracker(conversation_id)]
 
+    trackers = [get_events_before_restart(tracker) for tracker in trackers]
```
Comment (ancalita) [inline rasa/server.py:324]: Why is this required, does `Restarted` presence in `tracker.events` cause a bug?

**Review by ancalita — COMMENTED** [review]
Great stuff 💯 Left a few clarifying questions 😄 
Also worth looking into the JSON removal and why it's causing the `test_evaluate_invalid_intent_model_file` to fail.

File: rasa/server.py (hunk 3 above)
Comment (melindaloubser1) [reply rasa/server.py:1164]: I'm not sure why, but it was removed already previously - it just wasn't documented that it was. Even passing a JSON body, it would get interpreted as YAML and throw an error.

File: rasa/server.py (hunk 4 above)
Comment (melindaloubser1) [reply rasa/server.py:324]: The presence of `Restarted` means the session is translated into a blank story  - due to using the applied events as defined [here](https://github.com/RasaHQ/rasa/blob/dc4930d50b800c6a059f87b3f92f27468d995aad/rasa/shared/core/trackers.py#L490-L491). Tbh, I think the more correct thing to do is to change the definition of how sessions are split to include restarted events at the beginning of each session instead of leaving them at the end - but I'm concerned this would result in changes elsewhere that we don't want.

File: rasa/shared/core/trackers.py (hunk 2 above)
Comment (melindaloubser1) [reply rasa/shared/core/trackers.py:963]: I'll take another look and add a test

File: rasa/server.py (hunk 4 above)
Comment (ancalita) [reply rasa/server.py:324]: I see, so this becomes a problem further below in L340 because `tracker.as_story()` makes a call to `applied_events` which will be an empty list if the last event was `Restarted`?

Comment (ancalita) [reply rasa/server.py:324]: What would happen if other events follow the `Restarted` event?

File: docs/docs/http-api.mdx [hunk 5]
```diff
@@ -50,54 +50,44 @@ The [SocketIO channel](./connectors/your-own-website.mdx#websocket-channel) does
...
-between docker containers).
+We recommend that you don't expose the Rasa Server to the outside world directly, but
+rather connect to it via e.g. Nginx.
 
 Nevertheless, there are two authentication methods built in:
 
 ### Token Based Auth
 
-Pass in the token using `--auth-token thisismysecret` when starting
+To use a plaintext token to secure your server, specify the token in the argument `--auth-token thisismysecret` when starting
```
Comment (ancalita) [inline docs/docs/http-api.mdx:60]: These are great additions, however I just realised that I didn't clarify in my ticket that the page that is missing instructions is the [API spec page ](https://rasa.com/docs/rasa/pages/http-api)🙈

File: rasa/server.py (hunk 3 above)
Comment (ancalita) [reply rasa/server.py:1164]: That's strange, according to this [deprecation warning](https://rasa.com/docs/rasa/next/migration-guide/#nlu-json-format) in the migration guide, the JSON nlu format should only be removed in 4.0 🤔

File: docs/docs/http-api.mdx (hunk 5 above)
Comment (melindaloubser1) [reply docs/docs/http-api.mdx:60]: ahhh I see I see, will further update the spec

File: rasa/server.py (hunk 4 above)
Comment (melindaloubser1) [reply rasa/server.py:324]: > so this becomes a problem further below in L340 because tracker.as_story() makes a call to applied_events which will be an empty list if the last event was Restarted?


Exactly.


> What would happen if other events follow the Restarted event?


That should not be possible in this context because `Restarted` is always followed directly by `SessionStarted`, and the conversation has already been split on `SessionStarted` events.

File: rasa/server.py (hunk 3 above)
Comment (melindaloubser1) [reply rasa/server.py:1164]: Ah right I don't mean it was removed everywhere, but it was removed for this endpoint specifically.

Comment (melindaloubser1) [reply rasa/server.py:1164]: If you think it should still be supported, then I'll revert this and fix the handling of JSON to work

Comment (ancalita) [reply rasa/server.py:1164]: I think we should be consistent and support it here too, maybe worth adding a deprecation warning that this format will be removed from this endpoint in 4.0.

File: rasa/shared/core/trackers.py (hunk 2 above)
Comment (melindaloubser1) [reply rasa/shared/core/trackers.py:963]: After looking at it again, I'm removing the fix. I think it requires more thought about what constitutes a "session" in the context of test stories. For example, should we be adding a record of the user's `/restart` utterance in test stories? What about `action_restart`? I think it's better to open a separate ticket and make decisions about these explicitly.

File: docs/docs/http-api.mdx (hunk 5 above)
Comment (melindaloubser1) [reply docs/docs/http-api.mdx:60]: I added some description to the spec page, lmk if it serves the purpose you had in mind

File: rasa/server.py (hunk 3 above)
Comment (melindaloubser1) [reply rasa/server.py]: have added a method for it.

File: docs/docs/http-api.mdx (hunk 5 above)
Comment (ancalita) [reply docs/docs/http-api.mdx:60]: Looks great 🙌🏼

File: rasa/server.py [hunk 6]
```diff
@@ -1484,6 +1518,18 @@ def _validate_yaml_training_payload(yaml_text: Text) -> None:
         )
 
 
+def _validate_json_training_payload(json_text: Text) -> None:
+    try:
+        RasaYAMLReader
```
Comment (ancalita) [inline rasa/server.py:1523]: This appears incomplete 😄 also probably not needed, since you used `rasa.shared.utils.validation.validate_training_data` function

File: rasa/server.py [hunk 7]
```diff
@@ -1472,6 +1479,33 @@ def _training_payload_from_yaml(
     )
 
 
+def _nlu_training_payload_from_json(
```
Comment (ancalita) [inline rasa/server.py:1482]: Could we have a unit test covering this new function? 🙏🏼

**Review by ancalita — COMMENTED** [review]
Almost there 💯 Only asked for one unit test and probably some code can be removed.

File: rasa/server.py (hunk 6 above)
Comment (melindaloubser1) [reply rasa/server.py:1523]: ope this is entirely irrelevant sorry

**Comment by melindaloubser1** [comment]
Follow up issue is here https://github.com/RasaHQ/rasa/issues/11151

File: rasa/server.py [hunk 8]
```diff
@@ -1152,7 +1157,9 @@ async def evaluate_intents(
                     test_data, config_file, int(cross_validation_folds)
                 )
         else:
-            test_data = _test_data_file_from_payload(request, temporary_directory)
+            payload = _nlu_training_payload_from_json(request, temporary_directory)
```
Comment (ancalita) [inline rasa/server.py:1160]: Should we add the deprecation warning here too?

Comment (melindaloubser1) [reply rasa/server.py:1160]: Fortunately it appears automatically because the method calls other methods that contain it

**Review by ancalita — APPROVED** [review]
🎉
